nservicebus/security/encryption.md

```diff
@@ -43,7 +43,40 @@ You can also use the `WireEncryptedString` type to flag that a property should b
 
 ### Enabling property encryption
 
-Property encryption is enabled via the configuration API.
+Property encryption is enabled via the configuration API. 
+
+#### Key identifier
+
+Each key needs an unique key identifier (`KeyIdentifier`). The key identifier is communicated in the message header meta data and provides the receiving endpoint information on which key to use for decryption.
+
+NOTE: If a key identifier is not set then no messages can be send but received messages without a key identifier header will be decrypted using the keys that have no or an empty KeyIdentifier set.
+
+> Encrypted message has no 'NServiceBus.KeyIdentifier' header. Possibility of data corruption. Please upgrade endpoints that send messages with encrypted properties.
+
+NOTE: Key identifiers are supported since v3.3.16+, v4.7.8+, v5.0+ and newer. All support decrypting messages that have encrypted fragments but no key identifier header.
+
+
+#### Multiple key identifiers
+
+There can be environments where multiple senders share the same key but have different key identifiers. This is supported by seperating the key identifiers with a semi colon `;`.
+
+NOTE: The `;` character is reserved as a seperator and cannot be used.
+
+
+### Using the same key with and without a key identifier
+
+If the KeyIdentifier attribute is set then this key will not be used to decrypt message that do not have a key identifier header to prevent a false positive while decrypting as this would result in message corruption.
+
+Add an *empty* string (0 length) identification to the KeyIdentifier attribute in order to use a key for decrypting messages with and without key identifier meta info.
+
+
+#### Key format
+
+The key format can be specfied in either ASCII or Base64 format.
+
+With ASCII its not possible to use the full 8bit range of a byte as its a 7bit encoding and even then some characters need to be escaped which is not done resulting in even less characters. Meaning per byte only about 100 values are used. When you use 16 byte / 128 bit keys this means only about 100^16 combinations are possible. 
+If you use a 
+ An alternative is to specifify the key as Base64 encoded string.
```

Comment:
Looks like something got screwy here on Line 78?

Also, I would lead with Base64 as it is preferred. Right now it sounds like an also-ran. Make it clear that ASCII is not preferred and is included for backwards compatibility only.